alvr/server/src/connection.rs

```diff
@@ -748,19 +770,66 @@ fn try_connect(mut client_ips: HashMap<IpAddr, String>) -> ConResult {
                 if let Some(stats) = &mut *STATISTICS_MANAGER.lock() {
                     stats.report_tracking_received(tracking.target_timestamp);
 
+                    // Handle hand gestures
+                    if let Switch::Enabled(controllers_config) = &settings.headset.controllers {
+                        if let Switch::Enabled(gestures_config) =
+                            &controllers_config.hand_tracking.use_gestures
+                        {
+                            let mut hand_gesture_manager_lock = hand_gesture_manager.lock();
+
+                            if tracking.hand_skeletons[0].is_some() {
+                                trigger_hand_gesture_actions(
+                                    gestures_button_mapping_manager.as_mut().unwrap(),
+                                    *LEFT_HAND_ID,
+                                    &hand_gesture_manager_lock.get_active_gestures(
+                                        tracking.hand_skeletons[0].unwrap(),
+                                        gestures_config.clone(),
+                                        *LEFT_HAND_ID,
+                                    ),
+                                );
+                            }
+                            if tracking.hand_skeletons[1].is_some() {
+                                trigger_hand_gesture_actions(
+                                    gestures_button_mapping_manager.as_mut().unwrap(),
+                                    *RIGHT_HAND_ID,
+                                    &hand_gesture_manager_lock.get_active_gestures(
+                                        tracking.hand_skeletons[1].unwrap(),
+                                        gestures_config.clone(),
+                                        *RIGHT_HAND_ID,
+                                    ),
+                                );
+                            }
+
+                            drop(hand_gesture_manager_lock);
+                        }
+                    }
+
+                    let mut hand_skeletons_enabled = false;
+                    if let Switch::Enabled(controllers) = &config.controllers {
+                        hand_skeletons_enabled = controllers.hand_tracking.enable_skeleton;
+                    }
```

Comment:
You can simplify this to a if-else without using a mutable variable.

Reply:
Or you could use the functional patters. You can use `Switch::as_option()`and then use `Option` methods.